sdk/core/Azure.Core.TestFramework/src/TestEnvironment.cs

```diff
@@ -348,6 +353,7 @@ public void SetRecording(TestRecording recording)
         {
             _credential = null;
             _recording = recording;
+            IsInRecordedTest = true;
```

Comment:
either this or we make prop public or we reflect in tests like https://github.com/Azure/azure-sdk-for-net/blob/0c42a52f4a5385aa9487341a84b52d0587048b51/sdk/core/Azure.Core/tests/TestEnvironmentTests.cs#L123-L124 or we make internals visible to core tests.

Reply:
hmm, or refactor these tests...

Reply:
nvm.